src/rpdk/core/data/managed-upload-infrastructure.yaml

```diff
@@ -19,6 +19,31 @@ Resources:
             Status: Enabled
             AbortIncompleteMultipartUpload:
               DaysAfterInitiation: 1
+      VersioningConfiguration:
+        Status: Enabled
+      LoggingConfiguration:
+        DestinationBucketName: { 'Fn::Sub' : "artifact-bucket-access-logs-${AWS::AccountId}-${AWS::Region}" }
```

Comment:
This should probably be !Ref AccessLogsBucket ?
- Will setting this work if the AccessLogBucket does not exist?
- Will setting this work if the permissions are wrong (if not, it will also need a DependsOn)

Reply:
@ikben you are right, fixed.
- It will not work if AccessLogBucket does not exist
- also will not work if permissions are wrong
no need for DependsOn

Reply:
I think we might be using two different version for "works". I was talking about getting no errors in CloudFormation. If you can't put the LoggingConfiguration without having the permissions already there the dependsOn is needed, I assume this is not the case.

Reply:
You are right. Would you prefer to have CFN failing incase the policy does not exist then? dependsOn will add this constraint, correct?

Reply:
DependsOn will make sure that the AccessLogsPolicy is created before CloudFormation attempts to create the ArtifactBucket, so that the policy will always exist before the LoggingConfiguration is created / updated

Reply:
> DependsOn will make sure that the AccessLogsPolicy is created before CloudFormation attempts to create the ArtifactBucket, so that the policy will always exist before the LoggingConfiguration is created / updated

Does that work though? The Policy needs to `!Ref` the Bucket ARN for a bunch of settings?

Reply:
I just noticed that `AccessControl: LogDeliveryWrite` is in the Properties of `AccessLogsBucket`. I assume this is to make it possible to write to itself. If I understand this correctly than there are two options to make the template work:

- Keep the `AccessControl` and remove the whole `AccessLogsPolicy` as both do the same thing
- Remove the `AccessControl` and add the `DependsOn`. This will only be possible when not writing to itself (see other comment)

Reply:
> Does that work though? The Policy needs to `!Ref` the Bucket ARN for a bunch of settings?

This is how I read the template

```
ArtifactBucket -------------Ref-----------> AccessLogsBucket
    \---Depend---> AccessLogsPolicy ---Sub-->/
```
That does not create a circle (AccessLogsBucket does not need to know about ArtifactBucket)